api/v1alpha1/shared_types.go

```diff
@@ -627,33 +627,48 @@ type ResponseOverride struct {
 // CustomResponseMatch defines the configuration for matching a user response to return a custom one.
 type CustomResponseMatch struct {
 	// Status code to match on. The match evaluates to true if any of the matches are successful.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=50
 	StatusCode []StatusCodeMatch `json:"statusCode"`
```

Comment:
can we rename this to `statusCodes` ?

Reply:
Done. We may also want to rename the `ResponseOverride` in the BTP spec.

```
	// ResponseOverride defines the configuration to override specific responses with a custom one.
	// If multiple configurations are specified, the first one to match wins.
	//
	// +optional
	ResponseOverrides []*ResponseOverride `json:"responseOverrides,omitempty"`
```

Reply:
We've historically named all the top level fields  as singular names